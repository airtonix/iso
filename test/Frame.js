const React = require('react')
const { create: render } = require('react-test-renderer')
const Frame = require('../src/Frame').default

describe('Frame', () => {
  test('renders', () => {
    const json = render(
      React.createElement(Frame, {},
        React.createElement('h1', null, 'hello')
      )
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with overlay', () => {
    const json = render(
      React.createElement(Frame, { overlay: true },
        React.createElement('h1', null, 'hello')
      )
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders without children', () => {
    const json = render(
      React.createElement(Frame, {})
    ).toJSON()
    expect(json).toMatchSnapshot()
  })
})
