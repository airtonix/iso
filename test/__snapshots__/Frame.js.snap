// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame renders 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <iframe
    onLoad={[Function]}
    scrolling="yes"
    srcDoc="<style> *{box-sizing:border-box} body{margin:0} body{min-height:100vh} body{font-family:-apple-system,BlinkMacSystemFont,sans-serif;line-height:1.5}</style>
<div id='app'></div>"
    style={
      Object {
        "backgroundColor": "#fff",
        "border": 0,
        "display": "block",
        "height": "calc(100vh - 40px)",
        "margin": 0,
        "opacity": 1,
        "overflow": "scroll",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Frame renders with overlay 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <iframe
    onLoad={[Function]}
    scrolling="yes"
    srcDoc="<style> *{box-sizing:border-box} body{margin:0} body{min-height:100vh} body{font-family:-apple-system,BlinkMacSystemFont,sans-serif;line-height:1.5}</style>
<div id='app'></div>"
    style={
      Object {
        "backgroundColor": "#fff",
        "border": 0,
        "display": "block",
        "height": "calc(100vh - 40px)",
        "margin": 0,
        "opacity": 1,
        "overflow": "scroll",
        "width": "100%",
      }
    }
  />
  <div
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2,
      }
    }
  />
</div>
`;

exports[`Frame renders without children 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <iframe
    onLoad={[Function]}
    scrolling="yes"
    srcDoc="<style> *{box-sizing:border-box} body{margin:0} body{min-height:100vh} body{font-family:-apple-system,BlinkMacSystemFont,sans-serif;line-height:1.5}</style>
<div id='app'></div>"
    style={
      Object {
        "backgroundColor": "#fff",
        "border": 0,
        "display": "block",
        "height": "calc(100vh - 40px)",
        "margin": 0,
        "opacity": 0.25,
        "overflow": "scroll",
        "width": "100%",
      }
    }
  />
</div>
`;
